### ISSUE TITLE
UDF File System Repair - fsck

### ISSUE THREAD

**Issue by stef204** [pali/udftools#48]
Hi,
I used your `udftools` to create 2 partitions on an external USB drive. My intention was to use a format which would be recognized by Linux, Windows and*or MacOS; including permissions, etc.

I attempted to download a very large source code repository to the drive, and had to interrupt the download.

It left some "files" which are marked:

`??????????? somefilename` which are impossible to remove.

`rm -rf somefilename` returns `cannot remove input output error` and I am led to believe this means the file system is in need of repair? (Something like fsck,)

I cannot find any tools to repair  udffs.  Any suggestions?

**Comment by pali** [comment]
Duplicate of #21

**Comment by stef204** [comment]
Not totally the same, though.  Do you have any suggestions with regard to the second part of my post, how to resolve the problem?

**Comment by pali** [comment]
If filesystem is damaged it is not possible to mount and use it. So you need to repair it and after that you can remove any content in that which you wish.

**Comment by stef204** [comment]
Yes, HOW do I repair it, that's what I am asking.  I CAN mount it on my Linux box and navigate to the files marked:
`??????????? somefilename` but then I just CANNOT remove those files.  That's where I get the I/O error.
I am trying to mount the drive on a Windows VM now to see if I can use the `chkdsk` there but having trouble with getting the drive detected and read.
Can you suggest any way of fixing the issue described using Linux?

**Comment by pali** [comment]
That is why I closed this issue as duplicate of the #21 which is about fsck.udf tool for repairing udf filesystem. There is no released tool yet, only older WIP.

**Comment by stef204** [comment]
So you have no feedback or suggestions as to HOW to resolve the problem described with tools available currently?  You are saying there is no solution available for me?  Or can you suggest anything?

**Comment by pali** [comment]
As you can see #21 is still unresolved. I have already wrote that there is no released tool yet. You can only try mentioned tool from WIP pull request.

**Comment by stef204** [comment]
FYI, I was able to mount it and have it detected by Windows7. I ran the `chkdsk /f /r /x` to no avail.  It did not fix anything.  Just for the record.
I guess I will nuke the partitions(only has some backups on it that I can re-create) and start fresh. Unfortunately, I will have to drop UDF in favor of a more mature file system.  Hopefully, though, I can revisit UDF when it reaches a more mature stage.  Thanks.
